bson/primitive/primitive.go

```diff
@@ -215,3 +233,24 @@ type M map[string]interface{}
 //
 // 		bson.A{"bar", "world", 3.14159, bson.D{{"qux", 12345}}}
 type A []interface{}
+
+// RecursiveMap creates a map from the elements of the A, recursively creating
+// maps from embedded documents and arrays within A.
+func (a A) RecursiveMap() M {
```

Comment:
A `RecursiveMap()` function doesn't make sense for a slice type. Any conversion of an `A` should output another slice.

It seems this function is here to simplify conversion of `D` into `M` recursively in `D.RecursiveMap()`. That should be accomplished by iterating through the elements in an `A`, converting any `D` types to an `M`, and inserting the values back into an `A`.